GamepadEventDrivenInputAPI/explainer.md

```diff
@@ -0,0 +1,251 @@
+# Event-Driven Gamepad Input API
+
+## Authors:
+
+- [Sneha Agarwal](https://github.com/snehagarwal_microsoft)
+- [Steve Becker](https://github.com/SteveBeckerMSFT)
+- [Gabriel Brito](https://github.com/gabrielsanbrito)
+
+## Participate
+- [Gamepad API input events #662](https://github.com/w3ctag/design-reviews/issues/662)
+
+## Status of this Document
+
+This document is a starting point for engaging the community and standards bodies in developing collaborative solutions fit for standardization. As the solutions to the problems described in this document progress along the standards-track, we will retain this document as an archive and use this section to keep the community up-to-date with the most current standards venue and content location of future work and discussions.
+
+- This document status: **Active**
+- Expected venue: **[W3C Web Applications Working Group](https://www.w3.org/groups/wg/webapps/)**
+- Current version: **This document**
+
+## Introduction
+
+This explainer proposes an event-driven Gamepad Input API for the web, designed to complement the existing polling-based model. By enabling input events to be dispatched in response to changes in gamepad state, this API aims to support low-latency scenarios such as cloud gaming, where timely and reactive input delivery is critical.
+
+This proposal builds on earlier work by Chromium engineers, which explored event-driven gamepad input handling. (Note: The original proposal is documented in a [Google Doc](https://docs.google.com/document/d/1rnQ1gU0iwPXbO7OvKS6KO9gyfpSdSQvKhK9_OkzUuKE/edit?pli=1&tab=t.0).)
+
+## Definitions
+
+### Input Frame: 
+Each input frame refers to a single timestamped update of a gamepad’s state, typically derived from a HID (Human Interface Device) report, including all button and axis values at that moment in time.
+
+### RawGamepadInputChangeEvent: 
+An event that represents a snapshot of a gamepad’s state at the moment a new input frame is received from the gamepad device. Each event corresponds to a full input report (e.g., a HID report) and contains the complete state of all buttons, axes. This event enables applications to react to input in a timely, event-driven manner, as an alternative to polling via navigator.getGamepads().
+
+## User-Facing Problem
+
+The Gamepad API lacks event-driven input handling, requiring applications to poll for input state changes. This polling model makes it difficult to achieve low-latency responsiveness, as input changes can be missed between polling intervals. Developers working on latency-sensitive applications, such as cloud gaming platforms, have reported needing to poll at very high frequencies to detect input as quickly as possible. However, even with aggressive polling, scripts may still struggle to react in real time, especially under heavy UI thread load or on resource-constrained devices.
```

Comment:
I like how the original proposal gave a concrete example with metrics:

" If an application polls at a regular interval, the average added delay is equal to half the polling interval. For instance, if an application polls at 60 Hz then the polling interval is 16⅔ milliseconds and adds 8⅓ milliseconds average latency."

Should we add something similar to describe the latency problem?